src/Symfony/Component/Notifier/Bridge/Twitter/TwitterTransport.php

```diff
@@ -0,0 +1,292 @@
+
+        foreach (\is_array($options['body'] ?? null) ? $options['body'] : [] as $v) {
+            if (!$v instanceof DataPart) {
+                continue;
+            }
+            $formDataPart = new FormDataPart($options['body']);
+            foreach ($formDataPart->getPreparedHeaders()->all() as $header) {
+                $options['headers'][] = $header->toString();
+            }
+            $options['body'] = $formDataPart->bodyToIterable();
+            break;
+        }
+
+        $oauth = [
+            'oauth_consumer_key' => $this->apiKey,
+            'oauth_nonce' => self::$nonce = md5(self::$nonce ??= random_bytes(16)),
+            'oauth_signature_method' => 'HMAC-SHA1',
+            'oauth_timestamp' => time(),
+            'oauth_token' => $this->accessToken,
+            'oauth_version' => '1.0',
+        ];
+
+        $sign = $oauth + ($options['query'] ?? []) + (\is_array($options['body'] ?? null) ? $options['body'] : []);
+        ksort($sign);
+
+        $oauth['oauth_signature'] = base64_encode(hash_hmac(
+            'sha1',
+            implode('&', array_map('rawurlencode', [
+                $method,
+                $url,
+                implode('&', array_map(fn ($k) => rawurlencode($k).'='.rawurlencode($sign[$k]), array_keys($sign))),
+            ])),
+            rawurlencode($this->apiSecret).'&'.rawurlencode($this->accessSecret),
+            true
+        ));
+
+        $options['headers'][] = 'Authorization: OAuth '.implode(', ', array_map(fn ($k) => $k.'="'.rawurlencode($oauth[$k]).'"', array_keys($oauth)));
+
+        return $this->client->request($method, $url, $options);
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof ChatMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, ChatMessage::class, $message);
+        }
+
+        $options = $message->getOptions()?->toArray() ?? [];
+        $options['text'] = $message->getSubject();
+
+        if (isset($options['attach'])) {
+            $options['media']['media_ids'] = $this->uploadMedia($options['attach']);
+            unset($options['attach']);
+        }
+
+        $response = $this->request('POST', '/2/tweets', ['json' => $options]);
+
+        try {
+            $statusCode = $response->getStatusCode();
+        } catch (TransportExceptionInterface $e) {
+            throw new TransportException('Could not reach the remote Twitter server.', $response, 0, $e);
+        }
+
+        $result = $response->toArray(false);
+
+        if (400 <= $statusCode) {
+            throw new TransportException($result['title'].': '.($result['errors'][0]['message'] ?? $result['detail']), $response);
+        }
+
+        $sentMessage = new SentMessage($message, (string) $this);
+        $sentMessage->setMessageId($result['data']['id']);
+
+        return $sentMessage;
+    }
+
+    private function uploadMedia(array $media): array
```

Comment:
I would add the phpdoc for the type. This way, the psalm job can identify mistakes:

```php
/**
 * @param array<array{file: File, alt: string, subtitles: File|null, category: string|null, owners: string[]}> $media
 */
```